dev-client/src/util.ts

```diff
@@ -69,3 +69,15 @@ export const removeKeys = (a: any, b: any) => {
     }
   }
 };
+
+export const normalizeText = (text: string) =>
+  text
+    .trim()
+    .toLowerCase()
+    .normalize('NFD')
+    .replace(/[\u0300-\u036f]/g, ''); // unicode range for combining diacritical marks
+
+export const searchText = (needle: string) => (haystack?: string) =>
+  haystack !== undefined ? normalizeText(haystack).includes(needle) : false;
+
+export const equals = (a: any) => (b: any) => a === b;
```

Comment:
question: why does this exist?

Reply:
It's for the types of the filter algorithms that compare for equality.